project/MimaExcludes.scala

```diff
@@ -66,6 +66,12 @@ object MimaExcludes {
     ProblemFilters.exclude[Problem]("org.apache.spark.sql.catalyst.*"),
```

Comment:
Do we need to remove this exclusion in order to be able to detect incompatibilities in Catalyst?

Reply:
Oh wait, I see now: code under the `catalyst` module isn't necessarily under the Catalyst package, so after this patch we'll be able to check those classes. 👍

Reply:
this was the exactly same confusion I had .. maybe we would have to change the name or refactor with least invasive way .....